core/charm/repository/charmhub.go

```diff
@@ -389,40 +387,19 @@ func (c *CharmHubRepository) ListResources(charmURL *charm.URL, origin corecharm
 // a slice with the results. The results include the minimum set of metadata
 // that is required for deploying each charm.
 func (c *CharmHubRepository) GetEssentialMetadata(reqs ...corecharm.MetadataRequest) ([]corecharm.EssentialMetadata, error) {
```

Comment:
I believe you can delete lines 392 to and including 431. GetEssentialMetadata does batching which is no longer required as there are no macaroons to batch. getEssentialMetadataForBatch does the real work still required.

Then the TODO at 436 can be removed.